src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -4763,4 +4756,159 @@ public static OptionalDouble getSeverityDuringDecommission()
+    {
+        return ofNullable(confValueAccessors.get(name)).orElseThrow(() -> new PropertyNotFoundException(name))
+                                                       .get(conf);
+    }
+
+    public static String getStringProperty(String name)
+    {
+        Property property = ofNullable(confValueAccessors.get(name))
+                            .orElseThrow(() -> new PropertyNotFoundException(name));
+        return property.getType().equals(String.class) ?
+               (String) property.get(conf) :
+               propertyToStringConverter().convertNullable(property.get(conf));
+    }
+
+    /**
+     * Set configuration property for the given name to {@link #conf} if a safe manner
+     * with handling internal Cassandra exceptions.
+     *
+     * @param name Property name.
+     * @param value Property value.
+     */
+    public static void setProperty(String name, Object value)
+    {
+        try
+        {
+            Property property = ofNullable(confValueAccessors.get(name)).orElseThrow(() -> new PropertyNotFoundException(name));
+            if (property.getAnnotation(Mutable.class) == null)
+                throw new ConfigurationException("Property is read-only: " + property.getName());
+
+            if (value == null)
+            {
+                property.set(conf, null);
+                return;
+            }
+
+            if (property.getType().isAssignableFrom(value.getClass()))
+                property.set(conf, value);
+            else if (String.class.equals(value.getClass()))
+                property.set(conf, propertyFromStringConverter(primitiveToWrapper(property.getType())).convert((String) value));
+            else
+                throw new ConfigurationException(String.format("Cannot convert value '%s' to type '%s'.",
+                                                               value, property.getType().getSimpleName()), false);
+            logger.info("Property {} updated to {}", name, value);
+        }
+        catch (ConfigurationException e)
+        {
+            throw e;
+        }
+        catch (Exception e)
+        {
+            throw new ConfigurationException(e.getMessage(), false);
+        }
+    }
+
+    /**
+     * @return A converter that converts given object to a string.
+     * @param <T> Type to convert from.
+     */
+    public static <T> TypeConverter<T, String> propertyToStringConverter()
+    {
+        return obj -> YamlConfigurationLoader.YamlFactory.instance.newYamlInstance(Config.class).dump(obj).trim();
+    }
+
+    /**
+     * @param toClass Class to convert to.
+     * @return A converter that converts given string to an object of the given class.
+     * @param <R> Type to convert to.
+     */
+    public static <R> TypeConverter<String, R> propertyFromStringConverter(Class<R> toClass)
+    {
+        return obj -> YamlConfigurationLoader.YamlFactory.instance.newYamlInstance(toClass).loadAs(obj, toClass);
+    }
```

Comment:
this belongs in `SettingsTable`, this is leaking details to unrelated classes

Reply:
Moved to the SettingsTable.